primo/utils/config_utils.py

```diff
@@ -628,7 +635,18 @@ def _on_change(self, data) -> None:
         """
         self._text = data["new"]
 
-        self.widget.options = self.choices
+        if self.max_options_displayed is None:
+
+            self.widget.options = self.choices
+
+        else:
+            filtered = [well for well in self.choices if self._text in str(well)]
```

Comment:
Please replace variable name `well` with something agnostic like `choice` because presumably this widget can be applicable in other contexts